lib/json/JsonBonus.cpp

```diff
@@ -886,4 +886,19 @@ CSelector JsonUtils::parseSelector(const JsonNode & ability)
 	return ret;
 }
 
+std::shared_ptr<Bonus> JsonUtils::parseChargeBasedSpell(const JsonNode & ability, const std::function<void(const SpellID &)> & callback)
```

Comment:
Might be better to keep in art handler - this is not some generic functionality that can be reused, but rather part of charged artifact loading